### PR TITLE
Use Clang workaround for Thrust regarless of the version

### PR DESCRIPTION
**Pull request by dalg24**
I encountered the bug with CUDA-10.0 and LLVM/Clang v10.0.0

Rather than attempting to discriminate combinations of CUDA/Clang versions, I suggest we use the workaround unconditionally for now.

### CODE REVIEW

File: src/details/ArborX_DetailsSortUtils.hpp [hunk 1]
```diff
@@ -21,8 +21,8 @@
 
 // clang-format off
 #if defined(KOKKOS_ENABLE_CUDA)
-#  if defined(KOKKOS_COMPILER_CLANG) && KOKKOS_COMPILER_CLANG < 900
-// Clang of version less than 9.0 cannot compile Thrust, failing with errors
+#  if defined(KOKKOS_COMPILER_CLANG)
+// A range of versions of Clang cannot compile Thrust, failing with errors
```
Comment (aprokop) [inline src/details/ArborX_DetailsSortUtils.hpp:25]: Please specify at least which ones you tried.

Comment (dalg24) [reply src/details/ArborX_DetailsSortUtils.hpp:25]: See description above.  Not planning to include that in the comment.

Comment (aprokop) [reply src/details/ArborX_DetailsSortUtils.hpp:25]: I would prefer having a comment in the code, as I would not remember to look at github PR description. In general, I think github does not substitute the need for good description in commit messages nor code comments.

File: src/details/ArborX_DetailsSortUtils.hpp [hunk 2]
```diff
@@ -22,10 +22,14 @@
...
 #  if defined(KOKKOS_COMPILER_CLANG)
-// A range of versions of Clang cannot compile Thrust, failing with errors
-// like this:
+// Some versions of Clang fail to compile Thrust, failing with errors like
+// this:
 //    <snip>/thrust/system/cuda/detail/core/agent_launcher.h:557:11:
 //    error: use of undeclared identifier 'va_printf'
+// The exact combination of versions for Clang and Thrust (or CUDA) for this
+// failure was not investigated, however even very recent version combination
+// (Clang 10 and Cuda 10) demonstrated failure.
```
Comment (dalg24) [inline src/details/ArborX_DetailsSortUtils.hpp:31]:
```suggestion
// (Clang 10.0.0 and Cuda 10.0) demonstrated failure.
```

Comment (aprokop) [reply src/details/ArborX_DetailsSortUtils.hpp:31]: Yeah, I was not sure which ones you tried.

**Comment by aprokop** [comment]
retest this please.

**Pull request merged**